modules/gitops-installing-argocd-agent-using-helm.adoc

```diff
@@ -0,0 +1,153 @@
+// Module included in the following assembly:
+//
+// * argocd-agent-installation/gitops-installing-argocd-agent-using-helm.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="gitops-installing-argocd-agent-using-helm_{context}"]
+= Installing the Argo CD Agent by using a Helm chart
+
+Use the Helm chart provided by the Argo CD Agent project to install the Agent component on the workload cluster. Argo CD Agent supports two modes for installation, *managed* and *autonomous*.
+
+.Procedure
+
+. Install the Managed Agent.
+In *managed* mode, the Agent must communicate with the spoke Redis server. However, the default network policy created by the {gitops-title} Operator does not allow this. To enable communication, you must create a custom network policy.
+
+.. Create a custom network policy YAML file:
++
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: custom-agent-redis-network-policy
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: <ArgoCD Instance Name>-redis
+  ingress:
+    - ports:
+        - protocol: TCP
+          port: 6379
+      from:
+        - podSelector:
+            matchLabels:
+              app.kubernetes.io/name: argocd-agent-agent
+  policyTypes:
+    - Ingress
+----
+
+.. Apply the network policy by using the following command:
++
+[source,terminal]
+----
+$ oc apply -f network-policy.yaml -n "<AGENT_NAMESPACE>" --context "<AGENT_CONTEXT>"
+----
+
+.. After applying the network policy, install the managed Agent using Helm by using the following command:
++
+[source,terminal]
+----
+$ helm install argocd-agent "<HELM_REPO>" \
+ --version "<CHART_VERSION>" \
+ --set namespaceOverride="<AGENT_NAMESPACE>" \
+ --set agentMode="managed" \
+ --set server="<PRINCIPAL_ROUTE_DNS>" \
+ --set argoCdRedisSecretName="<ArgoCD_NAME>-redis-initial-password" \
+ --set argoCdRedisPasswordKey="admin.password" \
+ --set redisAddress="<ArgoCD_NAME>-redis:6379" \
+ --set tlsClientKeyPath="" \
+ --set tlsClientCertPath="" \
+ --set tlsRootCAPath="" \
+ --kube-context "<AGENT_CONTEXT>"
+----
++
+where:
+
+`<dns>`:: Specifies the hostname of the Principal service. Agents use this value to connect to the Principal's gRPC service. This should match the `spec.host` value of the Principal's route, or `status.loadBalancer.ingress.hostname` when the Principal service is of type `LoadBalancer`.
+
+. Install the Autonomous Agent.
+Use the following command to install the autonomous Agent using Helm:
++
+[source,terminal]
+----
+$ helm install argocd-agent "<HELM_REPO>" \
+ --version "<CHART_VERSION>" \
+ --set namespaceOverride="<AGENT_NAMESPACE>" \
```

Comment:
Remove `--set namespaceOverride (...) \` and replace with `--namespace <AGENT_NAMESPACE> \`